### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by audukehinde**
# In this project I save user data to local storage and:

-When the user changes the content of any input field, the data is saved to the local storage.
-When the user loads the page if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -3,23 +3,28 @@ const navMenu = document.querySelector('.nav-menu');
 const close = document.querySelector('.close');
 const open = document.querySelector('.fa-bars');
 const navLink = document.querySelectorAll('.nav-link');
+const form = document.getElementById('form');
```
Comment (GzimAsani) [inline main.js:6]: - [ ] When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page. Currently both your `full name` and `email` stay auto-populated but the `textarea` doesn't so kindly fix this

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @audukehinde    
Good job so far! Your portfolio looks great :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good use of gitflow :+1: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :thought_balloon: _


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 1 above)
Comment (audukehinde) [reply main.js:6]: Dear GzimAsani, thanks for your feedback. I checked my code again and the textarea is working fine as the rest. Please advise me on what to do. Thanks

Comment (GzimAsani) [reply main.js:6]: Hi @audukehinde, please ask for another code review. It could be that the problem was on my side.

Comment (audukehinde) [reply main.js:6]: Hello, I just checked on another system and it worked

File: main.js [hunk 2]
```diff
@@ -44,143 +47,133 @@ form.addEventListener('submit', (e) => {
...
+
+const addDataToLocalStorage = () => {
+  const myFormData = {
+    fullName: name.value,
+    email: mail.value,
+    message: msg.value,
+  };
+  localStorage.setItem('myFormData', JSON.stringify(myFormData));
+};
+form.addEventListener('change', addDataToLocalStorage);
```
Comment (TedLivist) [inline main.js]: - [ ] Your code works but not as expected. Your inputs only saved to localStorage after removing focus. But they should save while typing. Simply replace 'change' with 'input'.

**Review by TedLivist — CHANGES_REQUESTED** [review]
Hi @audukehinde,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 2 above)
Comment (audukehinde) [reply main.js]: Thanks Teddy

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @audukehinde,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]

**Pull request merged**